DSharpPlus.Interactivity/Extensions/ClientExtensions.cs

```diff
@@ -0,0 +1,78 @@
+        {
+            if (client.GetExtension<InteractivityExtension>() != null) throw new InvalidOperationException("Interactivity is already enabled for this client.");
+
+            configuration ??= new InteractivityConfiguration();
+            var extension = new InteractivityExtension(configuration);
+            client.AddExtension(extension);
+
+            return extension;
+        }
+
+        /// <summary>
+        /// Enables interactivity for each shard.
+        /// </summary>
+        /// <param name="client">The shard client to enable interactivity for.</param>
+        /// <param name="configuration">Configuration to use for all shards. If one isn't provided, default configuration values will be used.</param>        
+        /// <returns>A dictionary containing new <see cref="InteractivityExtension"/> instances for each shard.</returns>
+        public static async Task<IReadOnlyDictionary<int, InteractivityExtension>> UseInteractivityAsync(this DiscordShardedClient client, InteractivityConfiguration configuration = null)
+        {
+            var extensions = new Dictionary<int, InteractivityExtension>();
+            await client.InitializeShardsAsync().ConfigureAwait(false);
+
+            foreach (var shard in client.ShardClients.Select(xkvp => xkvp.Value))
+            {
+                var extension = shard.GetExtension<InteractivityExtension>() ?? shard.UseInteractivity(configuration);
+                extensions.Add(shard.ShardId, extension);
+            }
+
+            return new ReadOnlyDictionary<int, InteractivityExtension>(extensions);
+        }
+
+        /// <summary>
+        /// Retrieves the registered <see cref="InteractivityExtension"/> instance for this client.
+        /// </summary>
+        /// <param name="client">The client to retrieve an <see cref="InteractivityExtension"/> instance from.</param>
+        /// <returns>An existing <see cref="InteractivityExtension"/> instance, or <see langword="null"/> if interactivity is not enabled for the <see cref="DiscordClient"/> instance.</returns>
+        public static InteractivityExtension GetInteractivity(this DiscordClient client)
+            => client.GetExtension<InteractivityExtension>();
+
+        /// <summary>
+        /// Retrieves a <see cref="InteractivityExtension"/> instance for each shard.
+        /// </summary>
+        /// <param name="client">The shard client to retrieve interactivity instances from.</param>
+        /// <returns>A dictionary containing <see cref="InteractivityExtension"/> instances for each shard.</returns>
+        public static IReadOnlyDictionary<int, InteractivityExtension> GetInteractivity(this DiscordShardedClient client)
+        {
+            var extensions = new Dictionary<int, InteractivityExtension>();
+            client.InitializeShardsAsync().ConfigureAwait(false).GetAwaiter().GetResult();
```

Comment:
This reminds me from another PR, we should consider making this awaited and have the method be async, along with the same methods in the other extensions. cc @Emzi0767

Reply:
Yeah I agree. If we're making it async, we should make all the other methods async as well. While normally out-of-scope, I'll allow that for this PR.

Reply:
`DSharpPlus.Lavalink` and *VoiceNext* do not have an extension for `DiscordShardedClient`. However, both *CommandsNext* and *Interactivity* do. 
I've made the needed changes to the interactivity extension method in this PR, but I'd prefer to bring the other three DSharpPlus extensions in line with a dedicated PR.